src/openqaoa-core/openqaoa/backends/wrapper.py

```diff
@@ -49,6 +64,164 @@ def exact_solution(self, *args, **kwargs):
         return self.backend.exact_solution(*args, **kwargs)
 
 
+#------------------------ZNE WRAPPER---------------------------------
+available_factories = [
+    "Richardson",
+    "Linear",
+    "Poly",
+    "Exp",
+    "PolyExp",
+    "AdaExp",
+    "FakeNodes"
+]
+available_scaling = [
+    "fold_gates_at_random",
+    "fold_gates_from_right",
+    "fold_gates_from_left"
+]
+class ZNEWrapper(BaseWrapper):
+    """
+    This class inherits from the BaseWrapper and need to be backend agnostic
+    to QAOA implementations on different devices and their respectives SDKs. It 
+    implements Zero Noise Extrapolation (ZNE) from Mitiq framework. ZNE is an 
+    error mitigation technique used to extrapolate the noiseless expectation
+    value of an observable from a range of expectation values computed at 
+    different noise levels.
+
+    Parameters
+    ----------
+    factory: str
+        The name of the zero-noise extrapolation method. Supported values: "Richardson", "Linear", "Poly", "Exp", "PolyExp", "AdaExp", "FakeNodes".
+    scaling: str
+        The name of the function for scaling the noise of a quantum circuit. Supported values: "fold_gates_at_random", "fold_gates_from_right", "fold_gates_from_left".
+    scale_factors: List[int]
+        Sequence of noise scale factors at which expectation values should be measured.
+        For factory = "AdaExp", just the first element of the list will be considered.
+    order: int
+        Extrapolation order (degree of the polynomial fit). It cannot exceed len(scale_factors) - 1.
+        Only used for factory = "Poly" or "PolyExp".
+    steps: int
+        The number of optimization steps. At least 3 are necessary.
+        Only used for factory = "AdaExp".
+    """
+
+    def __init__(self, backend, factory, scaling, scale_factors, order, steps):
+        super().__init__(backend)
+
+        # only qiskit backends are supported
+        assert( type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.qasm_simulator'] or 
+                type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.shot_simulator'] or
+                type(backend) == DEVICE_NAME_TO_OBJECT_MAPPER['qiskit.statevector_simulator']
```

Comment:
I have noticed that we cannot use the statevector simulator with our current implementation. We used the get_counts to obtain the results which you cannot use with the statevector

Reply:
Addressed in the last commits